portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -18,76 +18,107 @@
-            response.sendError(response.SC_BAD_REQUEST, "Value entered in \'Set maximum number of comment\' is not a number!");
-            max_comments = defaultMaxComments;
-        }
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    ImmutableList<Entity> results =
+        ImmutableList.copyOf(
+            datastore
+                .prepare(query)
+                .asList(FetchOptions.Builder.withLimit(getMaxComments(request, response))));
 
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        ImmutableList<Entity> results = ImmutableList.copyOf(datastore.prepare(query).asList(FetchOptions.Builder.withLimit(max_comments))); 
+    List<Comment> comments = new ArrayList<>();
 
-        List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results) {
+      long id = entity.getKey().getId();
+      String message =
+          getTranslatedComment((String) entity.getProperty("message"), getLanguageCode(request));
+      long timestamp = (long) entity.getProperty("timestamp");
+      Comment comment = new Comment(id, message, timestamp);
+      comments.add(comment);
+    }
 
-        for (Entity entity : results) {
-            long id = entity.getKey().getId();
-            String title = (String) entity.getProperty("title");
-            long timestamp = (long) entity.getProperty("timestamp");
-            Comment comment = new Comment(id, title, timestamp);
-            comments.add(comment);
-        }
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.setCharacterEncoding("UTF-8");
+    response.getWriter().println(gson.toJson(comments));
+  }
 
-        Gson gson = new Gson();
-        response.setContentType("application/json;");
-        response.getWriter().println(gson.toJson(comments));
-    }
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if (request.getParameter("comment") != null) {
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("message", request.getParameter("comment"));
+      commentEntity.setProperty("timestamp", System.currentTimeMillis());
 
-    @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        Entity commentEntity = new Entity("Comment");
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(commentEntity);
+      response.sendRedirect("/index.html#add-comments");
+    } else response.sendError(response.SC_BAD_REQUEST, "Comment parameter missing");
+  }
 
-        if (request.getParameter("comment") == null)
-            response.sendError(response.SC_BAD_REQUEST, "Comment parameter missing");
+  private int getMaxComments(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    int max_comments = defaultMaxComments;
```

Comment:
I think the convention is to use `maxComments` for the variable. At least that would be consistent with other variables.

Reply:
Yes it makes sense, will do that.